# 2. Business Problem

A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.
Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.
File: LabTAT.csv


# Business Objective

The hospital wants to analyze whether there is a significant difference in the average Turn Around Time (TAT) for reports across four laboratories. If differences exist, the hospital can optimize lab selection and improve efficiency.

# Constraints

1. Data Availability – Accuracy depends on the quality of TAT data.
2. Normality & Variance Assumptions – Required for ANOVA; if violated, alternative tests are needed.
3. External Factors – Lab workload, sample type, and operational efficiency may impact TAT.
4. 5% Significance Level – Small probability of incorrect conclusions.


In [1]:
import pandas as pd
import numpy as np
import scipy 
from scipy import stats
#provides statistical functions
#stats contains a variety of statstical tests
from statsmodels.stats import descriptivestats as sd
#provides descriptive stastics tools, including the sign test
from statsmodels.stats.weightstats import ztest
#Used for conducting z-tests on datasets.
import pylab


In [2]:
data= pd.read_csv("lab_tat_updated.csv")
data.head()

Laboratory_1  Laboratory_2  Laboratory_3  Laboratory_4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [3]:
data.describe()

Laboratory_1  Laboratory_2  Laboratory_3  Laboratory_4
count    120.000000    120.000000    120.000000     120.00000
mean     178.257333    178.902917    200.210167     163.68275
std       13.919668     14.957114     15.794801      15.08508
min      140.250000    140.550000    170.580000     124.06000
25%      170.267500    168.025000    190.182500     154.05000
50%      179.055000    178.870000    198.610000     164.42500
75%      187.222500    189.112500    211.197500     172.88250
max      216.390000    217.860000    238.700000     205.18000

In [4]:
data.columns=['Laboratory_1','Laboratory_2','Laboratory_3','Laboratory_4']

In [5]:
data.isnull().sum()

Laboratory_1    0
Laboratory_2    0
Laboratory_3    0
Laboratory_4    0
dtype: int64

In [6]:
print("Shapiro_Lab1: ",stats.shapiro(data['Laboratory_1']))
print("Shapiro_Lab2: ",stats.shapiro(data['Laboratory_2']))
print("Shapiro_Lab3: ",stats.shapiro(data['Laboratory_3']))
print("Shapiro_Lab4: ",stats.shapiro(data['Laboratory_4']))


Shapiro_Lab1:  ShapiroResult(statistic=0.9886689843295583, pvalue=0.42316547647722136)
Shapiro_Lab2:  ShapiroResult(statistic=0.9936320020018462, pvalue=0.8637345311091214)
Shapiro_Lab3:  ShapiroResult(statistic=0.9796067665538031, pvalue=0.06547079619606683)
Shapiro_Lab4:  ShapiroResult(statistic=0.9913755409425613, pvalue=0.6619161995270413)


# Normality Result

All P-Values are greater than 0.05

Fails To reject the Null Hypothesis(Data is Normally Distributed)

# Variance Test

In [7]:
levene_test=scipy.stats.levene(data.Laboratory_1,data.Laboratory_2,data.Laboratory_3,data.Laboratory_4)
print('levene_test(Variance): ',levene_test)

levene_test(Variance):  LeveneResult(statistic=1.025294593220823, pvalue=0.38107781677304564)


# Variance Test Result 

pvalue=0.38107 > 0.05 that means Fails to reject Null Hypothesis(Accept H0)


# 1 way Anova test

In [10]:
anova_result=stats.f_oneway(data.Laboratory_1,data.Laboratory_2,data.Laboratory_3,data.Laboratory_4)
print("One Way Anova: ",anova_result)

One Way Anova:  F_onewayResult(statistic=121.39264646442368, pvalue=2.143740909435053e-58)


<b>#Result : P-value==> 2.1437

<b>#Interpretation:</b>
#H0: No significant difference in average TAT across labs.
#H1: There is significant diff in avg TAT across Labs.

#Since the p-value (2.1437)>0.05 that means Fails to reject the HO(Accept H0)

<b>#Conclusion: there is no significant difference in the average Turn Around Time (TAT) for reports across four laboratories.</b>